### PR TITLE
ref(docs/observability): updated observability docs

### PR DESCRIPTION
**Pull request by michelleN**
- reformatting / re-organizing
- modified wording

Signed-off-by: Michelle Noorali <minooral@microsoft.com>

Please mark with X for applicable areas.

-->
**Affected area**:

- New Functionality      [ ]
- Documentation          [ X]
- Install                [ ]
- Control Plane          [ ]
- CLI Tool               [ ]
- Certificate Management [ ]
- Networking             [ ]
- Metrics                [ ]
- SMI Policy             [ ]
- Security               [ ]
- Tests                  [ ]
- CI System              [ ]
- Demo                   [ ]
- Performance            [ ]
- Other                  [ ]

Please answer the following questions with yes/no.

- Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution? No

### CODE REVIEW

File: docs/content/docs/tasks_usage/metrics.md [hunk 1]
```diff
@@ -5,22 +5,48 @@ type: docs
 aliases: ["metrics.md"]
 ---
 
-
-# Metrics
-Open Service Mesh (OSM) generates detailed metrics for all services communicating within the mesh. These metrics provide insights into the behavior of services in the mesh helping users to troubleshoot, maintain and analyze their applications.
+## Table of Contents
```
Comment (nojnhuh) [inline docs/content/docs/tasks_usage/metrics.md:8]: The docs site generates and shows a table of contents for each doc on the right side of the page, so I'd be in favor of removing it here.

File: docs/content/docs/tasks_usage/metrics.md [hunk 2]
```diff
@@ -5,22 +5,48 @@ type: docs
...
+
+## Metrics Overview
+
+Open Service Mesh (OSM) generates detailed metrics related to all traffic within the mesh. These metrics provide insights into the behavior of applications in the mesh helping users to troubleshoot, maintain and analyze their applications.
 
 As of today OSM collects metrics directly from the sidecar proxies (Envoy). OSM provides rich metrics for incoming and outgoing traffic for all services in the mesh. With these metrics the user can get information about the overall volume of traffic, errors within traffic and the response time for requests.
 
-# Prometheus
-To facilitate gathering of consistent traffic metrics and statistics across all services in the mesh, OSM relies on [Prometheus][1]. Prometheus is an open-source monitoring and alerting toolkit which is commonly used (but not limited to) on Kubernetes and Service Mesh environments.
+OSM uses [Prometheus][1] to gather and store consistent traffic metrics and statistics for all applications running in the mesh. Prometheus is an open-source monitoring and alerting toolkit which is commonly used (but not limited to) on Kubernetes and Service Mesh environments.
```
Comment (nojnhuh) [inline docs/content/docs/tasks_usage/metrics.md:34]: Nit:
```suggestion
OSM uses [Prometheus][1] to gather and store consistent traffic metrics and statistics for all applications running in the mesh. Prometheus is an open-source monitoring and alerting toolkit which is commonly used on (but not limited to) Kubernetes and Service Mesh environments.
```

File: docs/content/docs/tasks_usage/metrics.md [hunk 3]
```diff
@@ -36,15 +62,17 @@ This command installs an osm control plane on the Kubernetes...
...
 ```
 
-### BYO (Bring-your-own)
+## Prometheus Integration
 
-The following section will document the additional steps needed to allow an already running instance to poll the endpoints of an OSM mesh.
+### BYO Prometheus
 
-#### List of Prerequisites
+The following section will document the additional steps needed to allow an already running prometheus instance to poll the endpoints of an OSM mesh.
```
Comment (SanyaKochhar) [inline docs/content/docs/tasks_usage/metrics.md:69]: nit:
```suggestion
The following section will document the additional steps needed to allow an already running Prometheus instance to poll the endpoints of an OSM mesh.
```

**Review by shashankram — APPROVED** [review]

File: docs/content/docs/tasks_usage/metrics.md [hunk 4]
```diff
@@ -166,18 +196,21 @@ Both metrics have the following labels:
...
+
 - HTTP requests that invoke a local response from Envoy have "unknown" `destination_*` labels on metrics.
     - In the demo, this includes requests from the bookthief to the bookstore.
 - Metrics are only recorded for traffic where both endpoints are part of the mesh. Ingress and egress traffic do not have statistics recorded.
 - Metrics are recorded in Prometheus with all instances of '-' and '.' in tags converted to '\_'. This is because proxy-wasm adds tags to metrics through the name of the metric and Prometheus does not allow '-' or '.' in metric names, so Envoy converts them all to '\_' for the Prometheus format. This means a pod named 'abc-123' is labeled in Prometheus as 'abc\_123' and metrics for pods 'abc-123' and 'abc.123' would be tracked as a single pod 'abc\_123' and only distinguishable by the 'instance' label containing the pod's IP address.
 
-## Querying metrics from Prometheus
+### Querying metrics from Prometheus
 
 ### Before you begin
```
Comment (SanyaKochhar) [inline docs/content/docs/tasks_usage/metrics.md:208]: I think this should be nested one level deeper?
```suggestion
#### Before you begin
```

File: docs/content/docs/tasks_usage/metrics.md [hunk 5]
```diff
@@ -166,18 +196,21 @@ Both metrics have the following labels:
...
+
 Ensure that you have followed the steps to run [OSM Demo][2]
 
 ### Querying proxy metrics for request count
```
Comment (SanyaKochhar) [inline docs/content/docs/tasks_usage/metrics.md:212]:
```suggestion
#### Querying proxy metrics for request count
```

**Review by nojnhuh — APPROVED** [review]
Left a couple nits, but looks good.

**Review by SanyaKochhar — APPROVED** [review]

**Comment by michelleN** [comment]
Removed TOC, addressed comments, fixed markdown linter issues.

**Pull request merged**